sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -117,31 +135,46 @@ def test_spectral_embedding_two_components(seed=36):
     true_label[0:n_sample] = 1
 
     se_precomp = SpectralEmbedding(
-        n_components=1, affinity="precomputed", random_state=np.random.RandomState(seed)
+        n_components=1,
+        affinity="precomputed",
+        random_state=np.random.RandomState(seed),
+        eigen_solver=eigen_solver,
     )
-    embedded_coordinate = se_precomp.fit_transform(affinity)
-    # Some numpy versions are touchy with types
-    embedded_coordinate = se_precomp.fit_transform(affinity.astype(np.float32))
-    # thresholding on the first components using 0.
-    label_ = np.array(embedded_coordinate.ravel() < 0, dtype="float")
-    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
+    for dtype in [np.float32, np.float64]:
+        embedded_coordinate = se_precomp.fit_transform(affinity.astype(dtype))
+        # thresholding on the first components using 0.
+        label_ = np.array(embedded_coordinate.ravel() < 0, dtype=np.float64)
```

Comment:
If we expect the label to be an integer, so the following will be fine.

```suggestion
        label_ = np.array(embedded_coordinate.ravel() < 0, dtype=np.int64)
```

Reply:
would this work in 32-bit OS? I have tried  similar changes suggested earlier by @ogrisel and found that tests fail in 32-bit OS

Reply:
I don't think that with integer we should have an issue here.